### PR TITLE
Documentation reviews

### PR DESCRIPTION
**Pull request by RainerKlute**

### CODE REVIEW

**Comment by andreasmuelder** [comment]
@RainerKlute @BeckmaR This could be conflicting with PR #1125

**Comment by BeckmaR** [comment]
I don'tthink the PRs have anything in common?

File: plugins/org.yakindu.sct.doc.user/src/user-guide/generating_code.textile [hunk 1]
```diff
@@ -1539,10 +1589,88 @@ It is the timer service's responsibility to actually raise a time event on a sta
...
 The @RuntimeService@ class maintains all state machines that are expected to execute run-to-completion steps periodically. A client application can retrieve the @RuntimeService@ singleton using @RuntimeService.getInstance()@. It can then pause, resume or cancel all state machines that are poised to run at a specified intervall.
-Note: This is only available with the _RuntimeService_ parameter of *GeneralFeatures* set to @true@, see "Java: General Features":#java-GeneralFeatures. 
+
+Please note: To enable this feature, set the _RuntimeService_ parameter of *GeneralFeatures* to @true@, see "Java: General Features":#java-GeneralFeatures. 
+
+
+
+h2(#creating-a-new-custom-code-generator-project). Creating a new custom code generator project
+
+Creating custom code generators is a first-level concept in YAKINDU Statechart Tools. You can develop custom code generators directly within your project workspace. You have the choice to choose "Xtend2":http://www.eclipse.org/xtend/ (preferred) or simply Java for implementing the code generator templates.
```
Comment (BeckmaR) [inline plugins/org.yakindu.sct.doc.user/src/user-guide/generating_code.textile]: The choice to choose? "you can choose between" or "your choices are" might be better. Furthermore, what's a first-level-concept meant to be? Even google didn't really help me. I'd stay with KISS and just say "is an important concept" or "interesting concept" or "built-in possibility" or whatever.

File: plugins/org.yakindu.sct.doc.user/src/user-guide/generating_code.textile [hunk 2]
```diff
@@ -1539,10 +1589,88 @@ It is the timer service's responsibility to actually raise a time event on a sta
...
 h3(#java-runtime-service). Runtime service
 
 The @RuntimeService@ class maintains all state machines that are expected to execute run-to-completion steps periodically. A client application can retrieve the @RuntimeService@ singleton using @RuntimeService.getInstance()@. It can then pause, resume or cancel all state machines that are poised to run at a specified intervall.
-Note: This is only available with the _RuntimeService_ parameter of *GeneralFeatures* set to @true@, see "Java: General Features":#java-GeneralFeatures. 
+
+Please note: To enable this feature, set the _RuntimeService_ parameter of *GeneralFeatures* to @true@, see "Java: General Features":#java-GeneralFeatures. 
+
+
+
+h2(#creating-a-new-custom-code-generator-project). Creating a new custom code generator project
```
Comment (BeckmaR) [inline plugins/org.yakindu.sct.doc.user/src/user-guide/generating_code.textile]: Is it _really_ useful to move this here? I'd link it but never move "Creating a new custom code generator project" out of the chapter "Custom code generators". That is confusing. Also, it's now a section in the chapter "Java Code generator"...?

**Comment by RainerKlute** [comment]
No, @andreasmuelder, there won't be any conflicts since the pull requests affect disjoint files.

File: plugins/org.yakindu.sct.doc.user/src/user-guide/generating_code_headless.textile [hunk 3]
```diff
@@ -1,53 +1,81 @@
...
 
-h2(#generating-state-machine-code-headless-installation). Installation
+###. FIXME: time-specific
 
-The headless code generator comes as an OSGi application which can be installed into every eclipse installation.
+bq.. *Please note:*
 
-If you download the "Professional Edition":https://info.itemis.com/yakindu/statecharts/pro the headless code generation feature is already included and no further installation is required.
-When you download the "Open Edition":https://info.itemis.com/download-yakindu-statechart-tools you can install the headless code generation feature from the SCT Pro update site.
+At the moment, headless code generation is available for the open-source generators only. This means generating source code for e.g. "Deep C Integration":../user-guide/c-domain.html#cdom_deep-c-integration statecharts or for your custom generators might fail. Support for C statecharts will be added with the next YAKINDU Statecharts release release.
```
Comment (BeckmaR) [inline plugins/org.yakindu.sct.doc.user/src/user-guide/generating_code_headless.textile]: "release release" - please fix double word

**Comment by andreasmuelder** [comment]
Perfect

**Pull request merged**